session-02/src/index.ts

```diff
@@ -0,0 +1,63 @@
+import fs from 'fs';
+import readline from 'readline';
+import os from 'os';
+import path from 'path';
+import { MyThreadPool } from './MyThreadPool';
+
+
+async function main() {
+  const inputFile = 'input.txt';
+  try {
+    console.log(`Reading numbers from "${inputFile}"…`);
+    const numbers = await readNumbers(inputFile);
```

Comment:
looks like the code reads all of the lines in the file into the RAM. if so, then the code is not scalable.